app/models/secret_sharing_grant.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+class SecretSharingGrant < ActiveRecord::Base
+  has_paper_trail skip: [:created_at]
+  belongs_to :project
+  validates :key, uniqueness: true
```

Comment:
I'm no rails/active-record expert, but this seems like possible cause of tripping up on uniqueness constraints, causing projects not being able to add secrets if one is already added for a key (no matter project) ?